test/common/run-tests

```diff
@@ -228,35 +229,67 @@ def run(opts, image):
     print("1..{0}".format(len(parallel_tests) + len(serial_tests)))
     flush_stdout()
 
+    if serial_tests and opts.list:
+        # Just build one batch for listing
+        batch_tests["list"] = {"working": None, "tests": [], "time": 0}
+        for test in serial_tests:
+            test.serial_machine = "list"
+            batch_tests["list"]["tests"].append(test)
+
     if serial_tests and not opts.list:
         if opts.machine:
+            batch_tests["theM"] = {"working": None, "tests": [], "time": 0}
```

Comment:
"theM" :grinning: More seriously, the indices could just be integers? That would also avoid the str(i) thingies

Reply:
as in "the Machine" :)
But yeah, I guess we can  just use integers, I used `str(i)` as I saw that you used `"1"` in global machines. I'll update it to use strings.